lib/get-project-context.js

```diff
@@ -0,0 +1,28 @@
+import escapeStringRegexp from "escape-string-regexp";
+import parseUrl from "parse-url";
+import urlJoin from "url-join";
+
+export default (
+  { envCi: { service } = {}, env: { CI_PROJECT_ID, CI_PROJECT_PATH } },
+  gitlabUrl,
+  gitlabApiUrl,
+  repositoryUrl
+) => {
+  const projectId = service === "gitlab" && CI_PROJECT_ID ? CI_PROJECT_ID : null;
+  const projectPath =
+    service === "gitlab" && CI_PROJECT_PATH
+      ? CI_PROJECT_PATH
+      : parseUrl(repositoryUrl)
+          .pathname.replace(new RegExp(`^${escapeStringRegexp(parseUrl(gitlabUrl).pathname)}`), "")
+          .replace(/^\//, "")
+          .replace(/\/$/, "")
+          .replace(/\.git$/, "");
+  const encodedProjectPath = encodeURIComponent(projectPath);
+  const projectApiUrl = urlJoin(gitlabApiUrl, `/projects/${projectId ?? encodedProjectPath}`);
+  return {
+    projectId,
```

Comment:
This is being exposed, but so far only being used in Unit Tests. It would be covered by the tests for `projectApiUrl`. Do we need to expose it?

Reply:
Good find, I think I used that in previous iterations. Will remove it!